src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java

```diff
@@ -431,6 +435,7 @@ public void endTransaction(){
         private final CloudioMessageFormat messageFormat;
         private final List<CloudioEndpointListener> listeners = new LinkedList<CloudioEndpointListener>();
         private String jobsFilePath;
+        private final Object transactionLock = new Object();
```

Comment:
No need for a dedicated object for synchronised, just do:

```java
synchronized(this){
    inTransaction = //...
}
```

Reply:
wasn't sure if _this_ was already used in some synchronized block, I'll change that then